components/teams-service/storage/postgres/postgres.go

```diff
@@ -362,24 +365,62 @@ func (p *postgres) PurgeUserMembership(ctx context.Context, userID string) ([]st
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
 func (p *postgres) AddUsers(ctx context.Context,
-	id string, userIDs []string) (storage.Team, error) {
-	var t storage.Team
-	err := p.db.QueryRowContext(ctx,
-		`WITH moved_row_ids AS (
-			INSERT INTO teams_users_associations (team_db_id, user_id, created_at)
+	id string, userIDs []string) (updatedUserIDs []string, err error) {
+
+	tx, err := p.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	projectsFilter, err := ProjectsListFromContext(ctx)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	var dbID int
+	var teamProjects []string
+	// ensure the team exists and isn't filtered out by the project filter
+	err = p.db.QueryRowContext(ctx,
+		`SELECT db_id, projects FROM teams
+		WHERE id = $1 AND projects_match(projects, $2::TEXT[])`,
+		id, pq.Array(projectsFilter)).
+		Scan(&dbID, pq.Array(&teamProjects)) // TODO is there a way to get these projects without returning them?
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	_, err = tx.ExecContext(ctx,
+		`INSERT INTO teams_users_associations (team_db_id, user_id, created_at)
 				SELECT db_id, unnest($2::TEXT[]), now()
-				FROM teams
-				WHERE id=$1
-			RETURNING team_db_id
-		)
-		UPDATE teams SET updated_at=NOW()
-		WHERE db_id in (SELECT DISTINCT * FROM moved_row_ids)
-		RETURNING id, name, projects, created_at, updated_at;`, id, pq.Array(userIDs)).Scan(
-		&t.ID, &t.Name, pq.Array(&t.Projects), &t.CreatedAt, &t.UpdatedAt)
+	 			FROM teams
+				WHERE db_id=$1
+				ON CONFLICT ON CONSTRAINT teams_users_pkey
```

Comment:
these lines, helpfully added by @vsingh-msys, handle the case when the user already exists on the team